### PR TITLE
add queries for missing resources

### LINKED ISSUE #13: add querys for missing resources
Add queries for missing resources:

- [ ] calendars
- [ ] coding-stations
- [ ] evva-components 
- [ ] office-modes
- [ ] time-profiles
- [ ] user (only queryById is missing)
- [ ] zones

Things To Do:
- [ ] add dataclass for resource (see https://integrations.api.xesar.evva.com/#operation-publish-xs3/1/q)
- [ ] add  "queryList" and "queryById" for each resource
- [ ] add test :)

### PR DESCRIPTION
**Pull request by berka3**
- added queries for resources (calendars, coding-stations, evva-components, office-modes, time-profiles, user, zones)
- added new Serializer (LocalDateSerializer, LocalTimeSerializer)
- changed some data types of existing resources

Closes #13

### CODE REVIEW

File: xesar-connect/src/test/kotlin/com/open200/xesar/connect/testutils/ZoneFixture.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.open200.xesar.connect.testutils
+
+import com.open200.xesar.connect.messages.query.Zone
+import java.util.*
+
+object ZoneFixture {
+
+    val zoneFixture =
+        Zone(
+            //installationPoint = listOf(InstallationPointFixture.installationPointFixture.id),
```
Comment (MartinPfeffer) [inline xesar-connect/src/test/kotlin/com/open200/xesar/connect/testutils/ZoneFixture.kt:10]: Can this line be removed?

**Review by MartinPfeffer — COMMENTED** [review]
Good work. :)

File: xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/Zone.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import com.open200.xesar.connect.utils.UUIDSerializer
+import kotlinx.serialization.Serializable
+import java.util.*
+
+@Serializable
+data class Zone(
+    val installationPoints: List<@Serializable(with = UUIDSerializer::class) UUID>? = null,
+    @Serializable(with = UUIDSerializer::class) val partitionId: UUID,
+    val installationPointCount: Int? = null,
+    val name: String? = null,
```
Comment (andreasgrill) [inline xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/Zone.kt:14]: I think these fields don't need to be optional.

File: xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/Zone.kt [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import QueryElementResource
+import QueryListResource
+import com.open200.xesar.connect.utils.UUIDSerializer
+import kotlinx.serialization.Serializable
+import java.util.*
+
+@Serializable
+data class Zone(
+    val installationPoints: List<@Serializable(with = UUIDSerializer::class) UUID>? = null,
```
Comment (andreasgrill) [inline xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/Zone.kt:11]: I think this field doesn't need to be optional, but rather an empty list.

File: xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/Calendar.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import QueryListResource
+import com.open200.xesar.connect.utils.LocalDateSerializer
+import com.open200.xesar.connect.utils.UUIDSerializer
+import kotlinx.serialization.Serializable
+import java.time.LocalDate
+import java.util.*
+
+@Serializable
+data class Calendar(
+    val serialVersionUID: Long? = null,
```
Comment (andreasgrill) [inline xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/Calendar.kt:13]: This is not needed, this is an error in the Xesar API documentation.

File: xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/Calendar.kt [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    @Serializable(with = UUIDSerializer::class) val partitionId: UUID? = null,
+    val name: String? = null,
+    val specialDays: List<@Serializable(with = LocalDateSerializer::class) LocalDate>? = null,
```
Comment (andreasgrill) [inline xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/Calendar.kt:16]: These fields should also be non-null.
The specialDays should be an empty list.

File: xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/CodingStation.kt [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.*
+
+@Serializable
+data class CodingStation(
+    @Serializable(with = UUIDSerializer::class) val id: UUID,
+    val name: String? = null,
+    val description: String? = null,
+    @Serializable(with = UUIDSerializer::class) val partitionId: UUID,
+    @Serializable(with = LocalDateTimeSerializer::class) val deleted: LocalDateTime? = null,
+    val online: Boolean? = null
```
Comment (andreasgrill) [inline xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/CodingStation.kt:18]: We need to check the fields on this and other resources if they are optional or required. I think they are similar to the Calendar where we have too many optional fields.

**Review by andreasgrill — COMMENTED** [review]
Great work for adding so many queries and improving the data types!
Maybe we need to check the resources if they are indeed optional most of the times. This is also a great opportunity to improve the Xesar API documentation.

File: xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/OfficeMode.kt [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+ * @param manualOfficeMode The manual office mode of the office mode.
+ * @param timeSeries The time series of the office mode.
+ * @param exceptionTimeSeries The exception time series of the office mode.
+ * @param timePointSeries The time point series of the office mode.
+ * @param exceptionTimePointSeries The exception time point series of the office mode.
+ */
+@Serializable
+data class OfficeMode(
+    val timeProfileDetails: String? = null,
+    val timeProfileName: String? = null,
```
Comment (andreasgrill) [inline xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/OfficeMode.kt:34]: I think the name could be non-null.

File: xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/OfficeMode.kt [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+import com.open200.xesar.connect.utils.LocalTimeSerializer
+import com.open200.xesar.connect.utils.UUIDSerializer
+import java.time.LocalTime
+import java.util.*
+import kotlinx.serialization.Serializable
+
+/**
+ * Represents an office mode in the system.
+ *
+ * @param id The unique identifier of the office mode.
```
Comment (andreasgrill) [inline xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/OfficeMode.kt:14]: For strange reasons the `id` is the installation point ID in this case, like the `installationPointId` property.

File: xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/Person.kt [hunk 9]
```diff
@@ -16,20 +16,21 @@ import kotlinx.serialization.Serializable
...
-    val firstName: String? = null,
-    val lastName: String? = null,
+    @Serializable(with = UUIDSerializer::class) val partitionId: UUID,
+    val firstName: String,
+    val lastName: String,
     val identifier: String? = null,
-    val personalReferenceDuration: PersonalReferenceDuration? = null,
-    val disengagePeriod: String? = null,
+    val personalReferenceDuration: PersonalReferenceDuration,
+    val disengagePeriod: String,
```
Comment (andreasgrill) [inline xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/Person.kt:33]: That could be mapped to an enum with `SHORT` and `LONG` values.

File: xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/User.kt [hunk 10]
```diff
@@ -22,14 +23,14 @@ import kotlinx.serialization.Serializable
...
-    val active: Boolean? = null,
+    val name: String,
+    val active: Boolean,
     val description: String? = null,
     @Serializable(with = LocalDateTimeSerializer::class) val lastActive: LocalDateTime? = null,
     @Serializable(with = LocalDateTimeSerializer::class) val lastLogin: LocalDateTime? = null,
     val loginIp: String? = null,
     val loginType: String? = null,
-) : QueryListResource {
+) : QueryListResource, QueryElementResource {
```
Comment (andreasgrill) [inline xesar-connect/src/main/kotlin/com/open200/xesar/connect/messages/query/User.kt:33]: I'm afraid there is no query support in Xesar for single users at the moment.

**Review by andreasgrill — COMMENTED** [review]
Amazing work! I've just a few suggestions.

**Review by andreasgrill — APPROVED** [review]

**Pull request merged**